internal/server/deviceflow.go

```diff
@@ -67,9 +67,14 @@ retry:
 // flow login. The response status can be pending, rejected, expired, or confirmed.
 func (a *API) GetDeviceFlowStatus(c *gin.Context, req *api.DeviceFlowStatusRequest) (*api.DeviceFlowStatusResponse, error) {
 	rctx := getRequestContext(c)
-	dfar, err := access.FindDeviceFlowAuthRequest(rctx, req.DeviceCode)
+	db := rctx.DBTxn
+
+	dfar, err := data.GetDeviceFlowAuthRequest(db, data.GetDeviceFlowAuthRequestOptions{ByDeviceCode: req.DeviceCode})
```

Comment:
Why skip the access layer here? I think the handler will end up just passing through (i.e. the call here is unauthenticated), but we probably want to keep call through access to maintain the idiom (since everything else calls through it).

Reply:
On second thought, this should skip the access layer because it's essentially the same thing as the login handler (which also skips the access layer).

Reply:
Definitely agree! I would go even further to say that all endpoints that do not require authorization should skip the access layer.

If the purpose of the access layer is "to notice when something is missing proper authorization", then a function in the `access` package that does not check authorization is actually defeating the purpose of this package.